Security_Engine/Create/CameraFieldOfView.cs

```diff
@@ -0,0 +1,364 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2022, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System.ComponentModel;
+using BH.oM.Geometry;
+using BH.oM.Security.Elements;
+using BH.oM.Base.Attributes;
+using System.Collections.Generic;
+using BH.Engine.Geometry;
+using System;
+using System.Linq;
+
+namespace BH.Engine.Security
+{
+    public static partial class Create
+    {
+        /***************************************************/
+        /****              Public Methods               ****/
+        /***************************************************/
+
+        [Description("Returns the polycurve that represents security camera field of view.")]
+        [Input("cameraDevice", "CameraDevice object to compute the field of view for.")]
+        [Input("obstacles", "Polyline objects that represents obstacles for camera vision (walls, columns, stairs).")]
+        [Output("PolyCurve", "PolyCurve object that represents security camera field of view.")]
+        public static PolyCurve CameraFieldOfView(this CameraDevice cameraDevice, List<Polyline> obstacles = null)
+        {
+            PolyCurve cameraCone = cameraDevice.CameraViewCone();
```

Comment:
We spoke about checking whether:
- the obstacles are planar/closed - fix with warning if not
- the obstacles are in parallel plane as camera (so parallel to global XY) - fix with warning if not
- the obstacles are in plane with camera - fix with a note if not

I guess it would be worth adding it somewhere at the beginning, or even a separate private method?